js/language_server/Makefile

```diff
@@ -2,9 +2,23 @@
 default: build
 
 include ../engine/Makefile.include
+include ../tree-sitter/Makefile.include
+
+# TODO: manage this better
+
+# All languages that have a C parser we need to compile w/emcc
+LANGUAGES := bash c cairo cpp csharp dart dockerfile go hack html java jsonnet kotlin lisp lua ocaml php promql protobuf python r rust solidity terraform typescript vue
+
+# Tree Sitter parser names so we can properly export them in EMCC
+TS_LANGS := bash c cairo clojure cpp c_sharp dart dockerfile go hack html java jsonnet kotlin lua ocaml php promql proto python r rust solidity hcl typescript vue
+
+empty:=
+space:= $(empty) $(empty)
+comma:=,
+PARSER_EXPORTED_FUNCTIONS := $(subst $(space),$(comma),$(TS_LANGS:%=_tree_sitter_%))
```

Comment:
please use sed or something else. This is ugly.

Reply:
So fair